### PR TITLE
Add full screen modal focus trap

### PR DESCRIPTION
**Pull request by jespino**
#### Summary
Add full screen modal focus trap

#### Ticket Link
[MM-17534](https://mattermost.atlassian.net/browse/MM-17534)

### CODE REVIEW

File: components/invitation_modal/invitation_modal.jsx [hunk 1]
```diff
@@ -184,13 +210,52 @@ export default class InvitationModal extends React.Component {
...
+                    id='invitation_modal.guests.title'
+                    defaultMessage='Invite **Guests**'
+                />
+            );
+        }
+
+        return (
+            <FormattedMarkdownMessage
+                id='invitation_modal.confirm.title'
+                defaultMessage='**People** Invited to **{teamName}**'
```
[the remark this answers is not in the document]
Comment (jespino) [reply components/invitation_modal/invitation_modal.jsx]: yes, a bad merge :(

File: components/invitation_modal/invitation_modal.jsx [hunk 2]
```diff
@@ -92,6 +105,19 @@ export default class InvitationModal extends React.Component {
         } else if (this.state.prevStep === STEPS_INVITE_MEMBERS) {
             this.setState({step: STEPS_INVITE_MEMBERS, prevStep: this.state.step, hasChanges: false, invitesSent: [], invitesNotSent: [], invitesType: InviteTypes.INVITE_MEMBER});
         }
+        if (this.modal && this.modal.current) {
+            this.modal.current.resetFocus();
+        }
+    }
+
+    getBackFunction = () => {
```
[the remark this answers is not in the document]
Comment (jespino) [reply components/invitation_modal/invitation_modal.jsx]: bad merge again

**Comment by srkgupta** [comment]
Hi @jespino 
Can you please resolve the merge conflicts. I am unable to test the same locally because of the merge conflicts.

**Comment by jespino** [comment]
@srkgupta it took me some time, but now should be properly merged.

**Comment by srkgupta** [comment]
Thanks @jespino

**Comment by jespino** [comment]
@srkgupta yes, that is related to the merge, I going to fix that.

**Comment by srkgupta** [comment]
Some more observations:
2. When the Invite People Modal is opened, it should be announced like any other dialogs (Manage Channels or Manage Team Members). Currently the JAWS reader does not reads anything when this is opened. 
3.  F6 or Ctrl+F6 still does not works on the Invite People modal. Currently it reads out the elements in the background of the modal. F6 or Ctrl+F6 should work like TAB in case of this modal. 
4. When someone selects the option to Invite Guests, JAWS reader should read out that the Invite Guests section has appeared. Currently it does not reads anything. 
5. JAWS reader reads out the Invite People, Search and Add Channels, Custom message input fields as "Edit-Text". It should read out the input fields with respective labels. For example: Add guests or email addresses input box. 
6. JAWS reader does not reads out the Autocomplete dropdowns properly. It tries to read the autocomplete results for each character typed and thereby it lags for the final character entered. 
7. Autocomplete options are not read out. JAWS reader just says to select options up or down, but it does not reads individual options in the Autocomplete options. It just says Option1 or Option2 selected, but does not reads out the actual option label. 
image

8. Discard Changes dialog is not read properly. It only reads `Discard Changes Dialog.`. It should read the full text also "Discard Changes Dialog? You have unsent invitations, are you sure you want to discard them?"

**Comment by jespino** [comment]
@srkgupta can you retry. The point 2 should be fixed, the point 3 maybe, and the rest I think is out of the scope of this concrete PR.

**Comment by srkgupta** [comment]
Removing the Cloud Test Server. Will need to confirm the scope of this PR with PM. Will pick it up once its confirmed.

**Comment by jespino** [comment]
@srkgupta I have fixed the point 2, the point 3 I have no idea how it works, probably we should defer it to @devinbinnie or @deanwhillier, I took a look of the previous F6 related changes, and I didn't found how to apply that to my PR, I think is not trivial and have relation with sections.

**Comment by srkgupta** [comment]
@jespino Some checks are unsuccessful. Can you please check, resolve and merge the PR.

**Pull request merged**